pyproject.toml

```diff
@@ -1,51 +1,27 @@
 [build-system]
-requires = ["setuptools>=61.2", "versioneer[toml]>=0.26"]
-build-backend = "setuptools.build_meta"
+requires = ["poetry-core"]
+build-backend = "poetry.core.masonry.api"
 
-[project]
+[tool.poetry]
 name = "py4ai-core"
+version = "v1.0.3-rc"
```

Comment:
I'm wondering if we should move away from `versioneer` and use poetry versioning built-in bumping feature, see [here](https://python-poetry.org/docs/cli/#version)

Reply:
Actually we moved away from versioneer too passing to [setuptools_scm](https://github.com/pypa/setuptools_scm) (which ofc is not suited for poetry). The main point I couldn't get about the tool you suggested is whether or not it is based on tags... How could we define a semi-automated version bump rule with that tool?

Reply:
The automation you have to implement (like in the e.g. CI/CD pipelines). But poetry gives you the ability to do update the version with `poetry version <BUMP_RULE>`, where the bump rule can be some of the keywords documented [here](https://python-poetry.org/docs/cli/#version)

Reply:
Uh, this is a little bit annoying... I've become quite lazy about implementing custom logic for version management... I was wondering if we should implement how to handle this detail before closing the PR, what do you think?